test/test_type_promotion.py

```diff
@@ -172,18 +172,19 @@ def test_mixed_type_backward(self, device):
                 func(x, y).sum().backward()
 
     def _get_test_tensor(self, device, dtype, remove_zeros=False):
-        shape = [20, 20, 20]
+        shape = [10, 10, 10]
```

Comment:
(carrying on from previous discussion) We should try to reuse https://github.com/pytorch/pytorch/blob/0febff36acc4a906ca95a9d39382028b10f92c33/test/test_torch.py#L14132-L14168 as much as possible instead of running our own implementation. It is generally very useful to be able to get a random test tensor.

Reply:
I'm still not entirely sold on this. Would you be OK refactoring to a common tensor generator as part of a separate PR? 

The _make_tensor version is far more simplistic, doesn't generate any negative values and few zeros, and has confusing handling of half tensors that I'd want to resolve before using outside test_torch. Fixing all that could mean changing a lot of existing tests unrelated to this PR.

Reply:
I think the best way to do it is to do the refactor in a PR and then have this diff depend on that PR.

> The _make_tensor version is far more simplistic, doesn't generate any negative values and few zeros, and has confusing handling of half tensors that I'd want to resolve before using outside test_torch. 

One alternative is to move your helper function to test_torch and have `make_tensor` call that.

I am proposing that we unify the idea of "getting a new random test tensor" and not duplicate the logic across the codebase. It sounds like to me your concern is about the mechanism in which we're combining the two functions; I haven't thought too much about what the right refactoring would look like.

Reply:
On second thought, you can do it as well in a follow-up PR. It would be good to file an issue about deduplicating the code and insert it as a comment in the code inline as a reminder.

Reply:
https://github.com/pytorch/pytorch/issues/30611